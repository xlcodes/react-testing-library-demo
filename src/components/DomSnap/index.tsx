import React, {FC} from "react";

interface IProps {
}

const DomSnap: FC<IProps> = ({}) => {

    return (
        <form aria-label={'form'}>
            <input
                type="text"
                name={'username'}
                disabled
                aria-disabled
                defaultValue={"xiaolin"}
                aria-label={'form_username'}
            />
            <input
                type={'number'}
                name={'age'}
                defaultValue={100}
                required
                aria-label={'form_age'}
            />
            <input
                type="radio"
                name={'sex'}
                value={'man'}
                defaultChecked
                aria-checked
                aria-label={'form_sex'}
            />
            <input
                type={'radio'}
                name={'sex'}
                value={'woman'}
                aria-label={'form_sex'}
            />
        </form>
    )
}

export default DomSnap