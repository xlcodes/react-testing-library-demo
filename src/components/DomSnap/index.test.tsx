import {render, screen} from "@testing-library/react";
import DomSnap from "./index";


describe("examples for snap", () => {
    it('a test for component snap', () => {
        const { baseElement } = render(<DomSnap />)
        expect(baseElement).toMatchSnapshot();
    })

    it('a test for part component snap', () => {
        render(<DomSnap />)
        expect(screen.getByRole('textbox', { name: "form_username" })).toMatchSnapshot();
    })

    it('a test for string snap', () => {
        expect('a test for string snap').toMatchSnapshot()
    })
})