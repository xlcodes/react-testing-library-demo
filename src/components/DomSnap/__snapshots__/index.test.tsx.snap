// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples for snap a test for component snap 1`] = `
<body>
  <div>
    <form
      aria-label="form"
    >
      <input
        aria-disabled="true"
        aria-label="form_username"
        disabled=""
        name="username"
        type="text"
        value="xiaolin"
      />
      <input
        aria-label="form_age"
        name="age"
        required=""
        type="number"
        value="100"
      />
      <input
        aria-checked="true"
        aria-label="form_sex"
        checked=""
        name="sex"
        type="radio"
        value="man"
      />
      <input
        aria-label="form_sex"
        name="sex"
        type="radio"
        value="woman"
      />
    </form>
  </div>
</body>
`;

exports[`examples for snap a test for part component snap 1`] = `
<input
  aria-disabled="true"
  aria-label="form_username"
  disabled=""
  name="username"
  type="text"
  value="xiaolin"
/>
`;

exports[`examples for snap a test for string snap 1`] = `"a test for string snap"`;
